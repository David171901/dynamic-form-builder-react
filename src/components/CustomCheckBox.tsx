import React from 'react';
import { ErrorMessage, useField } from 'formik';
import styles from '../styles/styles.module.css';

interface Props {
  label: string;
  name: string;
  [x: string]: any;
}

export const CustomCheckBox = (props: Props) => {
  const [field] = useField(props);

  return (
    <>
      <div>
        <div className={styles.checkbox_container}>
          <label className={styles.checkbox_label}>
            <input
              type="checkbox"
              {...field}
              {...props}
              className={styles.checkbox_input}
              checked={field.value}
            />
            {props.label}
          </label>
        </div>

        <ErrorMessage
          name={props.name}
          component="span"
          className={styles.error_field}
        />
      </div>
    </>
  );
};
