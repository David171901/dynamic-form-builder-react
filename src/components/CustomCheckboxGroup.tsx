import React from 'react';
import { ErrorMessage, Field, useField } from 'formik';
import styles from '../styles/styles.module.css';

type Opt = { value: string | number; desc: string };

interface Props {
  options: Opt[];
  name: string;
  label: string;
  [x: string]: any;
}

export const CustomCheckboxGroup = ({ label, options, ...props }: Props) => {
  const [field] = useField(props);

  return (
    <>
      <div>
        <label className={styles.label_field}>{label}</label>
        {options.map(opt => (
          <div className={styles.checkbox_container} key={opt.value}>
            <label className={styles.checkbox_label}>
              <Field
                {...field}
                {...props}
                type="checkbox"
                name={props.name}
                value={opt.value}
                className={styles.checkbox_input}
              />
              {opt.desc}
            </label>
          </div>
        ))}
        <ErrorMessage
          name={props.name}
          component="span"
          className={styles.error_field}
        />
      </div>
    </>
  );
};
