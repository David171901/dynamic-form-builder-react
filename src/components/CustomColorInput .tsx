import React from 'react';
import { useField } from 'formik';
import styles from '../styles/styles.module.css';

interface Props {
  name: string;
  label: string;
  placeholder?: string;
  [x: string]: any;
}
export const CustomColorInput = ({ label, ...props }: Props) => {
  const [field] = useField(props);

  return (
    <div>
      <label className={styles.label_field}>{label}</label>
      <input
        className={styles.color_select}
        {...field}
        {...props}
        type="color"
      />
    </div>
  );
};
