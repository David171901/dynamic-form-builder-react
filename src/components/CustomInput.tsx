import React from 'react';
import { ErrorMessage, useField } from 'formik';
import styles from '../styles/styles.module.css';

interface Props {
  name: string;
  label: string;
  type: string;
  placeholder?: string;
  [x: string]: any;
}

export const CustomTextInput = ({ label, ...props }: Props) => {
  const [field] = useField(props);

  return (
    <>
      <div>
        <label className={styles.label_field}>{label}</label>
        <input className={styles.input_field} {...field} {...props} />
        <ErrorMessage
          name={props.name}
          component="span"
          className={styles.error_field}
        />
      </div>
    </>
  );
};
