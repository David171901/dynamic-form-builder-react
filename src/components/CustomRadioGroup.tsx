import React from 'react';
import { useField, ErrorMessage } from 'formik';
import styles from '../styles/styles.module.css';

type Opt = { value: string | number; desc: string };

interface Props {
  options: Opt[];
  name: string;
  label: string;
  [x: string]: any;
}

export const CustomRadioGroup = ({ label, options, ...props }: Props) => {
  const [field] = useField(props);

  return (
    <>
      <div>
        <label className={styles.label_field}>{label}</label>
        {options.map(opt => (
          <div key={opt.value} className={styles.checkbox_container}>
            <label className={styles.checkbox_label}>
              <input
                {...field}
                {...props}
                type="radio"
                checked={opt.value === field.value}
                value={opt.value}
                className={styles.checkbox_input}
              />
              {opt.desc}
            </label>
          </div>
        ))}
        <ErrorMessage
          name={props.name}
          component="span"
          className={styles.error_field}
        />
      </div>
    </>
  );
};
