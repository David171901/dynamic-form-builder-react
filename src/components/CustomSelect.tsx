import React from 'react';
import { ErrorMessage, useField } from 'formik';
import styles from '../styles/styles.module.css';

interface Props {
  options: Opt[];
  label: string;
  name: string;
  [x: string]: any;
}

type Opt = { value: string | number; desc: string };

export const CustomSelect = ({ label, ...props }: Props) => {
  const [field] = useField(props);

  return (
    <>
      <div>
        <label className={styles.label_field}>{label}</label>
        <select {...field} {...props} className={styles.input_field}>
          <option value="">--- Select ---</option>
          {props.options.map(({ desc, value }) => (
            <option value={value} key={value}>
              {desc}
            </option>
          ))}
        </select>
        <ErrorMessage
          name={props.name}
          component="span"
          className={styles.error_field}
        />
      </div>
    </>
  );
};
