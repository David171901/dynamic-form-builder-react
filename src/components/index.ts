export * from './DynamicFormContainer'
export * from './Form'