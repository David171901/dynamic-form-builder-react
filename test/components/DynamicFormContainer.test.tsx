import React from 'react';
import renderer from 'react-test-renderer';
import { DynamicFormContainer, Form } from '../../src/components';
import { form } from '../data/form';

// const { act } = renderer;

describe('DynamicFormContainer', () => {
  test('Component', () => {
    const wrapper = renderer.create(
      <DynamicFormContainer formSchema={form} formTitle={'Form'}>
        {() => (
          <>
            <Form />
          </>
        )}
      </DynamicFormContainer>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
