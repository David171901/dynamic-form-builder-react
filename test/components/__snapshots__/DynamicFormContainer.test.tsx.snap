// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicFormContainer Component 1`] = `
<div>
  <h2
    className="text-2xl font-medium"
  >
    Form
  </h2>
  <form
    action="#"
    noValidate={true}
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div>
      <label
        className="label_field"
      >
        Full Name
      </label>
      <input
        className="input_field"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Full Name"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        className="label_field"
      >
        E-mail
      </label>
      <input
        className="input_field"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="E-mail"
        type="email"
        value=""
      />
    </div>
    <div>
      <label
        className="label_field"
      >
        Password
      </label>
      <input
        className="input_field"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="*******"
        type="password"
        value=""
      />
    </div>
    <button
      className="send_button"
      style={
        Object {
          "background": "#2980B9",
        }
      }
      type="submit"
    >
      SEND
    </button>
  </form>
</div>
`;
